src/Database/Query.php

```diff
@@ -1121,408 +1278,998 @@ private function get_search_sql( $string = '', $columns = array() ) {
+			'primary_alias'  => $this->table_alias,
+			'primary_column' => $this->get_primary_column_name(),
+			'meta_type'      => $this->get_meta_type(),
+			'query'          => $this
+		);
+
+		// Default values
+		$join = $where = array();
+
+		// Loop through parsers
+		foreach ( $parsers as $id ) {
+
+			// Skip
+			if ( empty( $id ) ) {
+				continue;
+			}
+
+			// Build the key
+			$key = strtolower( $id ) . '_query';
+
+			// Skip if no query vars
+			if ( empty( $query_vars[ $key ] ) || ! is_array( $query_vars[ $key ] ) ) {
+				continue;
+			}
+
+			// Add table alias to primary clause if not already set
+			if ( empty( $query_vars[ $key ][ 'alias'] ) ) {
+				$query_vars[ $key ][ 'alias'] = $args['table_alias'];
+			}
+
+			// Try to get the query var parser
+			$parser = $this->get_query_var_parser( $id, $query_vars[ $key ] );
+
+			// Skip if no query var parser
+			if ( empty( $parser ) ) {
+				continue;
+			}
+
+			// Default no subclauses
+			$subclauses = false;
+
+			// Set the key
+			$this->{$key} = $parser;
+
+			// Set the callback
+			$callback = array( $this->{$key}, 'get_sql' );
+
+			// Try to get the SQL subclauses
+			if ( is_callable( $callback ) ) {
+				$subclauses = call_user_func( $callback, array(
+					$args['meta_type'],
+					$args['primary_table'],
+					$args['primary_column'],
+					$args['query']
+				) );
+			}
+
+			// Skip if no SQL subclauses
+			if ( false === $subclauses ) {
+				continue;
+			}
+
+			// Set join
+			if ( ! empty( $subclauses['join'] ) ) {
+				$join[ $key ] = $subclauses['join'];
+			}
+
+			// Set where (removing " AND " from subclauses)
+			if ( ! empty( $subclauses['where'] ) ) {
+				$where[ $key ] = preg_replace( '/^\s*AND\s*/', '', $subclauses['where'] );
+			}
+		}
+
+		// Return join/where subclauses
+		return array(
+			'join'  => $join,
+			'where' => $where
+		);
+	}
+
+	/**
+	 * Parse a single query variable value.
+	 *
+	 * @since 2.1.0
+	 *
+	 * @param array  $query_vars
+	 * @param string $key
+	 *
+	 * @return int|string|array False if not set or default.
+	 *                          Value if object or array.
+	 *                          Attempts to parse a comma-separated string of
+	 *                          possible keys or numbers.
+	 */
+	private function parse_query_var( $query_vars = array(), $key = '' ) {
+
+		// Bail if no query vars exist for that ID
+		if ( ! isset( $query_vars[ $key ] ) ) {
+			return false;
+		}
+
+		// Get the value
+		$value = $query_vars[ $key ];
+
+		// Bail if equal to the exact default random value
+		if ( $value === $this->query_var_default_value ) {
+			return false;
+		}
+
+		/**
+		 * Early return objects, arrays, numerics, integers, or bools.
+		 *
+		 * These values assume the caller knew what it was doing, and simply
+		 * pass themselves through as "parsed" without any extra handling.
+		 */
+		if (
+			is_object( $value )
+			||
+			is_array( $value )
+			||
+			is_numeric( $value )
+			||
+			is_int( $value )
```

Comment:
All integer values are numeric. So we have to check it in reverse order.

```suggestion
			is_int( $value )
			||
			is_numeric( $value )
```

Discovered by @phpstan. Please consider using it.

Reply:
Nice find! Thank you.

I can't get PHPStan working locally; it keeps spitting out errors.

I'll keep tinkering with it though.